### PR TITLE
LPS-115851 Add horizontal scroll and better legend rendering

### PR DESCRIPTION
**Pull request by dgarciasarai**
**Description**

The main goal of this pull request is to show scroll if the graph doesn't fit the screen and stylish legend

**Changes**

On top of https://github.com/liferay-frontend/liferay-portal/pull/67

* 4112358: LPS-115851 Add horizontal scroll and better legend rendering

**Feature flag: YES**

The Audit Graph is hidden. The configuration file to show the chart is here: https://issues.liferay.com/browse/LPS-115805

**Module**

```
~/liferay-portal/modules/apps/content-dashboard/content-dashboard-web
```

**Screenshots**

Screenshot 2020-07-02 at 08 18 49

Screenshot 2020-07-02 at 08 18 57

cc: @bryceosterhaus @wincent

### CODE REVIEW

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
@wincent ready to review, only this commit 4112358

**Comment by wincent** [comment]
Looks fine to me, but I am not an expert on the usage of the layout classes, so I am going to assign to @eduardoallegrini because he's just done a bunch of those lately.

@eduardoallegrini: like @dgarciasarai says, https://github.com/liferay-frontend/liferay-portal/commit/4112358a6a7ae5e51b11fdf75ece189c4a66afcb is the commit to look at (the others have already been sent to Brian).

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js [hunk 1]
```diff
@@ -132,49 +125,102 @@ export default function AuditBarChart({rtl, vocabularies}) {
 	}, [colors]);
 
 	const renderLegend = (props) => {
-		const {payload} = props;
+		const {payload, yAxisName} = props;
+
+		return (
+			<ClayLayout.ContainerFluid>
+				<ClayLayout.Row justify="start">
```
Comment (eduardoallegrini) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js:132]: Isn't the row "justify-start" by default? Unless you had it overwritten somewhere else I think you can omit `justify="start"` here
https://codepen.io/pen?template=vMewQz

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js [hunk 2]
```diff
@@ -132,49 +125,102 @@ export default function AuditBarChart({rtl, vocabularies}) {
...
+					<ClayLayout.Col size={1}>
+						<span className="small">{yAxisName}:</span>
+					</ClayLayout.Col>
+					<ClayLayout.Col>
+						<ClayLayout.Row justify="start">
```
Comment (eduardoallegrini) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js:137]: as above

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js (hunk 1 above)
Comment (dgarciasarai) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js:132]: You're right. I'm going to remove it :)

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js (hunk 2 above)
Comment (dgarciasarai) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js:137]: 👍

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
@eduardoallegrini thanks for the review. Let's wait for the CI now 😄👍

**Comment by wincent** [comment]
Looks good to forward, and https://github.com/brianchandotcom/liferay-portal/pull/90855 is now merged too.

Not sure what will happen if we `ci:forward` though (seeing as the commits overlap). 🤔

**Comment by wincent** [comment]
I guess there's only one way to find out...

**Comment by wincent** [comment]
ci:forward

**Comment by wincent** [comment]
Sadly, forwarded PR is reporting merge conflicts, so it needs a rebase: https://github.com/brianchandotcom/liferay-portal/pull/90876

Back to you, @dgarciasarai.

**Comment by wincent** [comment]
I should note that the conflicts probably aren't even "real"; Brian cherry-picks commits which mean the hashes change. It's possible that GitHub isn't smart enough to realize that the commits are actually the same (ie. as per `git-patch-id`) as the ones already merged and can therefore be ignored.

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
🍀🤞  @wincent @eduardoallegrini

**Comment by dgarciasarai** [comment]
ci:forward

**Comment by wincent** [comment]
praying

**Pull request closed without merging**